pkg/cmd/cli/cmd/newapp.go

```diff
@@ -142,6 +142,7 @@ func NewCmdNewApplication(fullName string, f *clientcmd.Factory, out io.Writer)
 		},
 	}
 
+	cmd.Flags().BoolVar(&config.AsTestDeployment, "as-test", config.AsTestDeployment, "If true create this application as a test deployment, which validates that the deployment succeeds and then scales down.")
```

Comment:
It might make sense - it's one more set of arguments.  In our hypothesized split, it definitely belongs on `oc deploy` / `oc run`